pkg/alertmanager/alertstore/bucketclient/bucket_client.go

```diff
@@ -96,20 +96,27 @@ func (s *BucketAlertStore) SetAlertConfig(ctx context.Context, cfg alertspb.Aler
 		return err
 	}
 
-	return s.bucket.Upload(ctx, cfg.User, bytes.NewBuffer(cfgBytes))
+	// The specified object key is "" because the userID is injected by the user bucket client.
+	return s.getUserBucket(cfg.User).Upload(ctx, "", bytes.NewBuffer(cfgBytes))
```

Comment:
Interesting. It looks quite weird to be honest, and I would prefer if we didn't do it. If every operation in this file requires a comment, that is a design smell.